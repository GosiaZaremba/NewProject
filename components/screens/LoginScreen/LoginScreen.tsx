import React from 'react';
import { LoginTemplate } from '../../templates';
export type Props = {
    testID?: string;
};
export const LoginScreen: React.FC<Props> = ({ testID }) => {
    const onSubmitForm = (
        email: string,
        password: string,
        rememberMe: boolean | undefined
    ) => {
        console.log(
            'email',
            email,
            'password',
            password,
            'rememberMe',
            rememberMe
        );
    };
    const onPressFacebook = () => {
        console.log('Login with Facebook');
    };
    const onPressGoogle = () => {
        console.log('Login with Google');
    };

    const onPressForgotPassword = () => {
        console.log('Navigating to forgot password screen');
    };

    const onPressNavigateSignup = () => {
        console.log('Navigating to signup screen');
    };
    const onPressLinkedin = () => {
        console.log('Login with Linkedin');
    };

    return (
        <LoginTemplate
            onSubmitForm={onSubmitForm}
            onPressFacebook={onPressFacebook}
            onPressGoogle={onPressGoogle}
            onPressForgotPassword={onPressForgotPassword}
            onPressNavigateSignup={onPressNavigateSignup}
            onPressLinkedin={onPressLinkedin}
            testID={'loginScreen'}
        />
    );
};
