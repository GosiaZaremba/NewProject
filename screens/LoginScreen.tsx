import React from 'react';
import {LoginTemplate} from '../components/templates';

export type Props = {
  onSubmitForm: () => void;
  onPressGoogle: () => void;
  onPressFacebook: () => void;
  onPressForgotPassword: () => void;
  onPressNavigateSignup: () => void;
  onPressLinkedin: () => void;
};

export const LoginScreen: React.FC<Props> = () => {
  const onSubmitForm = () => {
    console.log('onSubmitForm');
  };
  const onPressFacebook = () => {
    console.log('onPressFacebook');
  };
  const onPressGoogle = () => {
    console.log('onPressGoogle');
  };

  const onPressForgotPassword = () => {
    console.log('onPressForgotPassword');
  };

  const onPressNavigateSignup = () => {
    console.log('onPressNavigateSignup');
  };
  const onPressLinkedin = () => {
    console.log('onPressLinkedin');
  };

  return (
    <LoginTemplate
      onSubmitForm={onSubmitForm}
      onPressFacebook={onPressFacebook}
      onPressGoogle={onPressGoogle}
      onPressForgotPassword={onPressForgotPassword}
      onPressNavigateSignup={onPressNavigateSignup}
      onPressLinkedin={onPressLinkedin}
    />
  );
};
