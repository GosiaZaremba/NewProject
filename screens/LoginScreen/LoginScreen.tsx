import React from 'react';
import {LoginTemplate} from '../../components/templates';

export type Props = {
  onSubmitForm: (
    email: string,
    password: string,
    rememberMe: boolean | undefined,
  ) => void;
  onPressGoogle: () => void;
  onPressFacebook: () => void;
  onPressForgotPassword: () => void;
  onPressNavigateSignup: () => void;
  onPressLinkedin: () => void;
};

export const LoginScreen: React.FC<Props> = () => {
  const onSubmitForm = (
    email: string,
    password: string,
    rememberMe: boolean | undefined,
  ) => {
    console.log('email', email, 'password', password, 'rememberMe', rememberMe);
  };
  const onPressFacebook = () => {
    console.log('onPressFacebook');
  };
  const onPressGoogle = () => {
    console.log('onPressGoogle');
  };

  const onPressForgotPassword = () => {
    console.log('onPressForgotPassword');
  };

  const onPressNavigateSignup = () => {
    console.log('onPressNavigateSignup');
  };
  const onPressLinkedin = () => {
    console.log('onPressLinkedin');
  };

  return (
    <LoginTemplate
      onSubmitForm={onSubmitForm}
      onPressFacebook={onPressFacebook}
      onPressGoogle={onPressGoogle}
      onPressForgotPassword={onPressForgotPassword}
      onPressNavigateSignup={onPressNavigateSignup}
      onPressLinkedin={onPressLinkedin}
    />
  );
};
