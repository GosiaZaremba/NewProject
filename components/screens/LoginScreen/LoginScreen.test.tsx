import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';
import 'react-native';
import { LoginScreen } from './LoginScreen';

describe('Screen LoginScreen', () => {
    test('Component Snapshot', () => {
        const tree = renderer.create(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('ensure login form works correctly', () => {
        const { getByTestId } = render(<LoginScreen />);

        const loginSwitch = getByTestId('organism-loginForm-switch');
        const inputPassword = getByTestId('organism-loginForm-password-input');
        const inputEmail = getByTestId('organism-loginForm-email-input');
        const submitButton = getByTestId('organism-loginForm-button');

        fireEvent.changeText(inputPassword, 'hasloLogin');
        fireEvent.changeText(inputEmail, 'login@wp.pl');
        fireEvent(loginSwitch, 'onValueChange', { value: true });
        fireEvent.press(submitButton);
    });

    test('ensure social login icons correctly', () => {
        const { getByTestId } = render(<LoginScreen />);

        const googleIcon = getByTestId('templates-loginTemplate-google-icon');
        const facebookIcon = getByTestId(
            'templates-loginTemplate-facebook-icon'
        );
        const linkedInIcon = getByTestId(
            'templates-loginTemplate-linkedIn-icon'
        );

        fireEvent.press(googleIcon);
        fireEvent.press(facebookIcon);
        fireEvent.press(linkedInIcon);
    });

    test('ensure login screen links work correctly', () => {
        const { getByTestId } = render(<LoginScreen />);
        const forgotPasswordLink = getByTestId(
            'templates-loginTemplate-forgotPassword-link'
        );
        const signUpLink = getByTestId('templates-loginTemplate-signUp-link');

        fireEvent.press(forgotPasswordLink);
        fireEvent.press(signUpLink);
    });
});
