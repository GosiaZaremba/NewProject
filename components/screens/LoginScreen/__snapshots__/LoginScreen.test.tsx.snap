// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen LoginScreen Component Snapshot 1`] = `
<View
  style={
    Object {
      "padding": 40,
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontSize": 20,
        "fontWeight": "700",
      }
    }
  >
    LOGIN
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <View
      style={
        Object {
          "marginVertical": 5,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 2,
              }
            }
            testID="organism-loginForm-email-label"
          >
            Email
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="email"
            secureTextEntry={false}
            style={
              Object {
                "borderColor": "#908D8E",
                "borderRadius": 5,
                "borderWidth": 1,
                "padding": 10,
                "width": "100%",
              }
            }
            testID="organism-loginForm-email-input"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "marginBottom": 2,
              }
            }
            testID="organism-loginForm-password-label"
          >
            Password
          </Text>
          <TextInput
            onChangeText={[Function]}
            placeholder="password"
            secureTextEntry={true}
            style={
              Object {
                "borderColor": "#908D8E",
                "borderRadius": 5,
                "borderWidth": 1,
                "padding": 10,
                "width": "100%",
              }
            }
            testID="organism-loginForm-password-input"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
          "marginTop": 10,
        }
      }
    >
      <View>
        <RCTSwitch
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#908D8E"
          style={
            Object {
              "height": 31,
              "width": 51,
            }
          }
          testID="organism-loginForm-switch"
          thumbTintColor="#908D8E"
          tintColor="#908D8E"
          value={false}
        />
        <Text
          style={
            Object {
              "color": "black",
              "marginBottom": 2,
            }
          }
          testID="organism-loginForm-label"
        >
          Remember me?
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "marginVertical": 15,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="Login"
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="organism-loginForm-button"
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fc296f",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "black",
          "textDecorationLine": "underline",
        }
      }
      testID="templates-loginTemplate-forgotPassword-link"
    >
      Forgot Password?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginVertical": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "#908D8E",
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 1,
          "justifyContent": "center",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F5F0F2",
            "borderColor": "#908D8E",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 5,
            "position": "absolute",
          }
        }
      >
        <Text>
          OR
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
        "marginVertical": 10,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#e63039",
          },
        ]
      }
      testID="templates-loginTemplate-google-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#e63039",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="templates-loginTemplate-google-icon"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#39539b",
          },
        ]
      }
      testID="templates-loginTemplate-facebook-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#39539b",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="templates-loginTemplate-facebook-icon"
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={5}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 50,
            "borderWidth": 3,
            "height": 45,
            "justifyContent": "center",
            "margin": 10,
            "opacity": 0.5,
            "padding": 10,
            "width": 45,
          },
          Object {
            "borderColor": "#3872c3",
          },
        ]
      }
      testID="templates-loginTemplate-linkedIn-pressable"
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#3872c3",
              "fontSize": 12,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="templates-loginTemplate-linkedIn-icon"
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "marginBottom": 2,
        }
      }
    >
      Need an account? 
    </Text>
    <Text
      onPress={[Function]}
      style={
        Object {
          "color": "black",
          "textDecorationLine": "underline",
        }
      }
      testID="templates-loginTemplate-signUp-link"
    >
      SIGN UP
    </Text>
  </View>
</View>
`;
